pom.xml

```diff
@@ -997,6 +997,58 @@
                 <module>integrations</module>
             </modules>
         </profile>
+        <profile>
+          <id>jdk9-javadoc</id>
+          <!-- https://maven.apache.org/guides/introduction/introduction-to-profiles.html#Details_on_profile_activation -->
+          <!-- AND semantics; see https://issues.apache.org/jira/browse/MNG-4565 -->
+            <activation> 
+              <jdk>[9,10)</jdk>
+              <property>
+                  <name>!jdkToolchainVersion</name>
+              </property>
+            </activation>
+            <build>
+                <pluginManagement>
+                    <plugins>
+                        <plugin>
+                            <groupId>org.apache.maven.plugins</groupId>
+                            <artifactId>maven-javadoc-plugin</artifactId>
+                            <configuration>
+                                <additionalJOptions>
+                                    <!-- https://bugs.openjdk.java.net/browse/JDK-8184969 -->
+                                    <JOption>-Xold</JOption>
+                                </additionalJOptions>
+                            </configuration>
+                        </plugin>
+                    </plugins>
+                </pluginManagement>
+            </build>
+        </profile>
+        <profile>
+          <id>jdk9-toolchain-javadoc</id>
```

Comment:
is that profile necessary ? Ins't the first profile actived if using jdk9 via toolchain ?

Reply:
It is necessary, sadly.

The first profile, `jdk9-javadoc`, is active when the JDK running Maven (so no toolchains are involved yet) is 9 or greater, but not 10, _and_ when the `jdkToolchain` property is not set (activations have `AND` semantics as of Maven 3.2.2):
```
<activation> 
    <jdk>[9,10)</jdk>
    <property>
        <name>!jdkToolchainVersion</name>
    </property>
</activation>
```
It is responsible for making sure the `maven-javadoc-plugin` is configured properly when toolchains are _not_ being used and Maven was run under a JDK that is version 9 or greater, but not 10 or greater.

The second profile, `jdk9-toolchain-javadoc`, is active when the `jdkToolchainVersion` property is set to `9`:
```
<activation>
    <property>
        <name>jdkToolchainVersion</name>
        <value>9</value>
    </property>
</activation>
```
It is responsible for making sure the `maven-javadoc-plugin` is configured properly when toolchains _are_ being used.  The two profiles are mutually exclusive.